chain/epoch_manager/src/lib.rs

```diff
@@ -1464,73 +1470,105 @@ impl EpochManager {
         Ok(*self.epoch_id_to_start.get(epoch_id).unwrap())
     }
 
-    /// Get epoch info aggregator and update it to block info as of `last_block_hash`. If `epoch_id`
-    /// doesn't match the epoch id of the existing aggregator, re-initialize the aggregator.
-    /// If `copy_only` is true, then we clone what is in the cache. Otherwise we take the aggregator
-    /// from cache and invalidates the cache.
-    pub fn get_and_update_epoch_info_aggregator(
+    /// Updates epoch info aggregator to block as of `last_block_hash` if
+    /// necessary.  If `epoch_id` doesn’t match the epoch id of the existing
+    /// aggregator, reinitialise the aggregator.  If `store_update` is true,
+    /// save the aggregator state to the store if epoch has changed or every
+    /// AGGREGATOR_SAVE_PERIOD heights.
+    pub fn update_epoch_info_aggregator(
         &mut self,
         epoch_id: &EpochId,
         last_block_hash: &CryptoHash,
-        copy_only: bool,
-    ) -> Result<EpochInfoAggregator, EpochError> {
-        let epoch_info_aggregator_cache = if copy_only {
-            self.epoch_info_aggregator.clone()
-        } else {
-            self.epoch_info_aggregator.take()
-        };
-        let mut epoch_change = false;
-        let mut aggregator = if let Some(aggregator) = epoch_info_aggregator_cache {
-            aggregator
-        } else {
-            epoch_change = true;
-            self.store
-                .get_ser(ColEpochInfo, AGGREGATOR_KEY)
-                .map_err(EpochError::from)?
-                .unwrap_or_else(|| EpochInfoAggregator::new(epoch_id.clone(), *last_block_hash))
-        };
-        if &aggregator.epoch_id != epoch_id {
-            aggregator = EpochInfoAggregator::new(epoch_id.clone(), *last_block_hash);
-            epoch_change = true;
+        store_update: Option<&mut StoreUpdate>,
```

Comment:
You can get rid of the option here if we access self.epoch_info_aggregator directly in the two places I commented above.